### PR TITLE
Fixed issue 863. Uses the MEMORY_BASIC_INFORMATION64 struct and retur…

### PR DESCRIPTION
**Pull request by jhumble**
…ns a 64bit baseAddress when compiled for 64 bit windows

### CODE REVIEW

File: psutil/_psutil_windows.c [hunk 1]
```diff
@@ -2694,9 +2694,13 @@ static char *get_region_protection_string(ULONG protection) {
  */
 static PyObject *
 psutil_proc_memory_maps(PyObject *self, PyObject *args) {
+	#ifdef _WIN64
+		MEMORY_BASIC_INFORMATION64 basicInfo;
+	#else
+		MEMORY_BASIC_INFORMATION basicInfo;
+	#endif
```
Comment (giampaolo) [inline psutil/_psutil_windows.c]: Please indent the `ifdef`s properly (no spaces) and also the variable names should be indented like the others below

File: psutil/_psutil_windows.c [hunk 2]
```diff
@@ -2720,19 +2724,30 @@ psutil_proc_memory_maps(PyObject *self, PyObject *args) {
 
     while (VirtualQueryEx(hProcess, baseAddress, &basicInfo,
                           sizeof(MEMORY_BASIC_INFORMATION)))
+
```
Comment (giampaolo) [inline psutil/_psutil_windows.c]: unnecessary \n

File: psutil/_psutil_windows.c [hunk 3]
```diff
@@ -2720,19 +2724,30 @@ psutil_proc_memory_maps(PyObject *self, PyObject *args) {
...
-            py_tuple = Py_BuildValue(
-                "(kssI)",
-                (unsigned long)baseAddress,
-                get_region_protection_string(basicInfo.Protect),
-                mappedFileName,
-                basicInfo.RegionSize);
+            #ifdef _WIN64
+               py_tuple = Py_BuildValue(
+                  "(KssI)",
+                  (unsigned long long)baseAddress,
```
Comment (giampaolo) [inline psutil/_psutil_windows.c]: here you can use a single `ifdef` to enclose only the `(unsigned long / long)` logic: there's no need to re-define the whole `Py_BuildValue` block twice. Also please indent the ifdefs (no spaces)

**Comment by giampaolo** [comment]
I added some comments inline. Other than that, can you please also update HISTORY and CREDITS files? Thanks a lot.

**Comment by jhumble** [comment]
I made your suggested changes. Let me know if I missed anything. Thanks!

File: psutil/_psutil_windows.c [hunk 4]
```diff
@@ -2727,12 +2731,20 @@ psutil_proc_memory_maps(PyObject *self, PyObject *args) {
...
+#ifdef _WIN64
+		   py_tuple = Py_BuildValue(
+			  "(KssI)",
+
+#else
+		   py_tuple = Py_BuildValue(
+			  "(kssI)",
+			  (unsigned long)baseAddress,
+#endif
+			  (unsigned long long)baseAddress,
```
Comment (giampaolo) [inline psutil/_psutil_windows.c]: this looks wrong. Shouldn't this line be *inside* the ifdef statement?
You should do:

```


 +#ifdef _WIN64
 +         py_tuple = Py_BuildValue(
 +            "(KssI)",
 +            (unsigned long long)baseAddress,
 +#else
 +         py_tuple = Py_BuildValue(
 +            "(kssI)",
 +            (unsigned long)baseAddress,
 +#endif
              ...
```

File: HISTORY.rst [hunk 5]
```diff
@@ -17,6 +17,7 @@ Bug tracker at https://github.com/giampaolo/psutil/issues
 - #715: don't crash at import time if cpu_times() fail for some reason.
 - #717: [Linux] Process.open_files fails if deleted files still visible.
 - #724: [FreeBSD] virtual_memory().total is slightly incorrect.
+- #863: [Windows] memory_map truncates addresses above 32 bits
```
Comment (giampaolo) [inline HISTORY.rst]: Mmmm looks like you based off this PR from `jhumble:863-fix-64bit-memory_map`. Also looks like you're trying to merge this into `giampaolo:0-wiz-0-master` branch. You should make a PR whch is based off from psutil latest `master` branch instead.

Comment (giampaolo) [inline HISTORY.rst]: ...or maybe I'm wrong? Anyway, by looking at the HISTORY file I can see that is an old HISTORY file. 
If I hit "merge" I am not sure this is gonna be merged into psutil master branch.

**Comment by jhumble** [comment]
You're right on both. Sorry, I'm pretty new to pull requests here. I'll try to just delete this request and make a new one off of master.

**Pull request closed without merging**